### PR TITLE
select keys specifically for relevant generations CORE-8525

### PR DESCRIPTION
**Pull request by mmaxim**

### CODE REVIEW

File: go/chat/teams.go [hunk 1]
```diff
@@ -227,7 +238,7 @@ func (t *TeamsNameInfoSource) Lookup(ctx context.Context, name string, public bo
 	if err != nil {
 		return res, err
 	}
-	return t.makeNameInfo(ctx, team, public)
+	return t.makeNameInfo(ctx, team, nil, public)
```
Comment (maxtaco) [inline go/chat/teams.go]: do we ever call `#Lookup` on the way into chat?

**Comment by mmaxim** [comment]
Fails `TestTeamReset` for some reason...

File: go/chat/boxer.go [hunk 2]
```diff
@@ -275,26 +275,13 @@ func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, conv
 				boxed.ClientHeader.TlfPublic))), nil
 	}
 	keyMembersType := b.getEffectiveMembersType(ctx, boxed, conv.GetMembersType())
-	nameInfo, err := CtxKeyFinder(ctx, b.G()).FindForDecryption(ctx,
+	encryptionKey, err := CtxKeyFinder(ctx, b.G()).FindForDecryption(ctx,
 		tlfName, boxed.ClientHeader.Conv.Tlfid, conv.GetMembersType(),
 		conv.IsPublic(), boxed.KeyGeneration, keyMembersType == chat1.ConversationMembersType_KBFS)
 	if err != nil {
 		// Check to see if this is a permanent error from the server
 		return chat1.MessageUnboxed{}, b.detectKBFSPermanentServerError(err, tlfName)
```
Comment (mlsteele) [inline go/chat/boxer.go]: I suspect any `DecryptionKeyNotFoundError` should be wrapped in `NewPermanentUnboxingError` before returning from `UnboxMessage`. Could add it to `detectKBFSPermanentServerError` (which is due for a rename to kitchen sink).

File: go/chat/teams.go [hunk 3]
```diff
@@ -414,7 +432,7 @@ func (t *ImplicitTeamsNameInfoSource) makeNameInfo(ctx context.Context, team *te
 		return res, errors.New("blank TLF ID given")
 	}
 	res.CanonicalName = impTeamName.String()
-	if res.CryptKeys, err = getTeamKeys(ctx, team, public); err != nil {
+	if res.CryptKeys, err = getTeamCryptKeys(ctx, team, public); err != nil {
```
Comment (mlsteele) [inline go/chat/teams.go]: Is this still called all the time? Does NameInfo need keys on it? If refactoring chat is a pain we can cache the verification of keys on teams so getting all keys is fastish again.

File: go/chat/boxer.go (hunk 2 above)
Comment (mmaxim) [reply go/chat/boxer.go]: Yeah true, I'll look at that.

File: go/chat/teams.go (hunk 3 above)
Comment (mmaxim) [reply go/chat/teams.go]: I'm not quite done, I am going to refactor a little more and make this call very rare.

**Pull request merged**